### PR TITLE
Moar sonar

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Refactoring

### Description

Get rid of Sonar warnings about declaring JUnit 5 methods public.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Review by franvila — APPROVED** [review]
LGTM

File: integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java [hunk 1]
```diff
@@ -148,9 +148,9 @@ public void shouldPassThroughRecordUnchanged(KafkaCluster cluster, Admin admin)
         }
     }
 
+    @SuppressWarnings("java:S5841")
     @Test
-    @SuppressWarnings("java:S5841") // java:S5841 warns that doesNotContain passes for the empty case. Which is what we want here.
```
Comment (SamBarker) [inline integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java:152]: Wasn't intentional to remove the comment

**Review by k-wall — APPROVED** [review]
Lgtm. 
I think it might be better to land #514 before this, then deal with the merge conflicts on this PR. The reason I say that is this PR is mostly mechanical in nature, and will take much less mental energy than reasoning about the more complicated changes we've made on #514.

**Comment by SamBarker** [comment]
I'll talk to @robobario about it, I'm in no rush and agree this is largely mechanical so easier to deal with here but I don't want to leave this hanging around waiting.

**Pull request merged**